man/nixnote.1
.TH NIXNOTE "1" "January 2011" "nixnote 1.1"
.SH NAME
NixNote \- An open source Evernote client.
.SH SYNOPSIS
NixNote is an open source clone of Evernote. Evernote is a collection of software and services that allows users to collect, sort, tag and annotate notes and other miscellaneous information. Data may be stored locally and/or synchronized with Evernote servers so they can be accessed by other clients.
.SH OPTIONS
.IP "NN_NAME=<name>"
The name of this NixNote account. This can be used to create separate NixNote databases. Separate databases do not need to be connected to the same Evernote account.
.IP "NN_HOME=<path>" 
The default NixNote database is stored in ~/.nevernote. Specifying this value will change where the database 
is located.
.IP "NN_XMS=<amount>M" 
Minimum amount of heap memory (in megabytes). The default is 256M.
.IP "NN_XMX=<amount>M"
Maximum amount of heap memory (in megabytes). The default is 1024M.
.IP "NN_NEW_RATIO=<number>"
The ratio between young & old generations. This is used in heap management and garbage collection. The default is 1.
.IP "NN_GC_OPT=<value>"
This determines how Java garbage collection is performed.  Valid values are -XX:+UseParNewGC, -XX:+UseConcMarkSweepGC, and -XX:+UseParallelGC. The default is -XX:+UseParallelGC and enables parallel garbage collection.
.IP "NN_DEBUG=<value>"
Used to set Java debug options for problem diagnosis. These values are not normally set.

.SH AUTHORS
NixNote is developed by Randy Baumgarte <randy@fbn.cx>.
.PP
This manual page was written by Vincent Cheng <Vincentc1208@gmail.com> and Randy Baumgarte for the Debian project (and may be used by others).


changelog.txt
NixNote (1.1) stable; urgency=low
  * Any type of attachment can be saved regardless of premium status. 
  * Added the ability to include children in the results when selecting a 
    parent tag.  This does NOT change the search behavior, but just the tag 
    selection via the cursor.  It is disabled by default but can be enabled 
    under Edit/Preferences. 
  * Added the ability to automatically wildcard any searches.  This will 
    match the behavior of Evernote's clients.  It is disabled by default to 
    improve performance, but you can enable it under Edit/Preferences/Indexing. 
  * Added the ability to "pin" notes.  Pinned notes will remain in the list 
    regardless of the search or serlection criteria.
  * Added the ability to do a "quick link".  If you highlight text in a note 
    and choose the quicklink option it will search for a note with the 
    selected words in the title.  If a match is found it will create a link 
    to that note.  If multiple notes are found it will bring up a dialog 
    box to choose the correct one.
  * Added notebook specific sorting.  When changing notebooks it should 
    revert to the order that notebook was last sorted in. 
  * If a note is pinned or if the title color is changed, that note will 
    be marked as unsynchronized & will be sent to Evernote if that notebook 
    is synchronized.  If you ever then restore your database or synchronize 
    that note to another machine running NixNote, the title color & 
    pinned attribute will be restored.  This only impacts notes in a 
    synchronized notebook and only newly changed notes.  Any existing ones 
    will not be synchronized until the next time the note is changed.
  * Added very basic support for importing from Evernote export files.  
    Only text notes can be imported.  Images or attachments may not yet work
    properly.
  * Added a global setting to display the contents of all notes 
    right-to-left instead of left-to-right.  It can be set under 
    Edit/Preferences.  This will NOT change the way Evernote displays your 
    notes.  I can't control that.  
  * Corrected some places where it still said NeverNote instead of NixNote.
  * Corrected a bug where searching for words didn't work properly if you 
    prepended the word with *
  * Corrected a problem where the tags would be unhidden after a sync.
  * Corrected a problem where non-text characters would be corrupted when 
    editing the source.
  * Corrected a performance problem when editing large notes.
  * Corrected a problem where trailing spaces in a note title would not 
    be removed prior to saving
  * Corrected a problem where links in notes wouldn't synchronize if the 
    URL didn't start with the protocol (i.e. http:// or ftp://).
  * Corrected a problem where # could not be used in the file name.
  * Corrected a problem where sorting was by ASCII value rather than 
    alphabetically.
  * Corrected an encryption padding problem (thanks Alex!).  
  * Corrected an error where the debug message level wasn't being set 
    upon startup. 
  * Reorganized the menus to match Evernote a little better.
  * Some third party libraries have been upgraded.
 -- Randy Baumgarte <randy@fbn.cx>  Sat, 08 Oct 2011 19:51:36 -0400

NixNote (1.0) stable; urgency=low
  * Changed from NeverNote to NixNote
  * Added support for note linking. It is still feels a little clunky but it seems usable.
  * Added indexing options to restrict what is indexed and to permit special characters that won't be removed from words.  You need to reindex your database when these values change.
  * Cleaned up the Android HTML Entities fix. It is still disabled by default and has a few issues, but it seems to be a little better now.
  * Fixed a problem where the LaTeX editor wouldn't be shown under some circumstances.
  * Fixed a problem where opening the current note in an external window might cause recent edits to be lost.
  * Fixed a problem where linked notebooks were not authorizing properly.
  * Fixed a problem where linked notebooks would interrupt a sync if authority for the user was revoked.  It will now prompt on what should be done.
  * Fixed a problem where changes made to automatic import folders were not saved.
  * Fixed a problem where Cut was not Ctrl+C, but center text was.  Center text was changed to Ctrl+E.
  * Fixed a problem where encrypting the database resulted in problems with images & attachments and the index table.
  * Added the ability to edit a note's HTML source (VERY EXPERIMENTAL).
 -- Randy Baumgarte <randy@fbn.cx>  Sat, 08 Oct 2011 19:50:36 -0400  

NeverNote (0.98) stable; urgency=low
  * Added the ability to insert & edit LaTeX formulas.
  * Added the option to add a tag as a child of the currently selected tag.
  * Added better save/restore window state logic so the toolbar position will be remembered after restarting.
  * Altered default style to be Cleanlooks.
  * Added a default-cleanlooks theme.
  * Altered note counting logic.  You should notice a large jump in performance if you had a large number of tags and/or notebooks and (for me anyway) it starts a little faster.
  * Altered multiple tag selection behavior to match Evernote's clients.  Prior releases would display notes matching any selected tags.  It now displays a note only if a note matches all selected tags.  This can be changed back to the prior behavior via the Edit/Preferences dialog box.
  * Altered the default behavior of minimizing to the tray.  By default it no longer does this, but minimizes to the task bar.  You can still set it via Edit/Preferences
  * I think I have the worst of the shared notebook problems corrected.
  * Hopefully corrected a problem where it would hang on shutting down if you selected "Synchronize On Close".
  * Corrected a problem where note information could be lost if opening a note in an external window after first editing it in the normal NeverNote window.

nixnote.sh
#! /bin/sh

eval $1

###########################################
# Location variables.  Edit the variables #
# below to your specific installation.    #
# The ones below are examples only.       #
###########################################

NIXNOTE=$(cd `dirname $0` && pwd)
# NIXNOTE=/usr/share/nixnote

########################################
# Memory settings.  These can be tuned #
# to your specific needs.  The greater #
# the memory allocated the better      #
# your response may be, but the more   #
# resources the program will consume.  #
# Lower numbers may hurt performance   #
# but will reduce resource held by     #
# the program.  If you get errors      #
# that say "out of memory" you need    #
# to increase these values.            #
########################################
# Initial heap size
NN_XMS=256M
# Maximum heap size
NN_XMX=1024M

## The young generation
# the young generation will occupy 1/2 of total heap
NN_NEW_RATIO=1

## GC option
## recommend Incremental Low Pause GC for desktop apps 
NN_GC_OPT=-Xincgc
## recent multi-core CPU may show good performance
#NN_GC_OPT=-XX:+UseParNewGC
#NN_GC_OPT=-XX:+UseConcMarkSweepGC
## same as default
#NN_GC_OPT=-XX:+UseParallelGC

## debug
#NN_DEBUG=-agentlib:hprof=format=b
#NN_DEBUG=-agentlib:hprof=cpu=samples,format=a
#NN_DEBUG=-verbose:gc 

########################################
# This next variable is optional. It   #
# is only needed if you want to run    #
# multiple copies of NixNote under     #
# the same Linux user id.  Each        #
# additional copy (after the first)    #
# should have a unique name.  This     #
# permits the settings to be saved     #
# properly.  If you only want to run   #
# one copy under a single userid, this #
# can be commented out.                #
########################################
#NN_NAME="sandbox"  


#Do any parameter overrides
while [ -n "$*" ]
do
   eval $1
   shift
done


###################################################################
###################################################################
## You probably don't need to change anything below this line.   ##
###################################################################
###################################################################


#####################
# Setup environment #
#####################
NN_CLASSPATH=$NIXNOTE/nixnote.jar

NN_CLASSPATH=$NN_CLASSPATH:$NIXNOTE/lib/apache-mime4j-0.6.jar
NN_CLASSPATH=$NN_CLASSPATH:$NIXNOTE/lib/commons-codec-1.5.jar
NN_CLASSPATH=$NN_CLASSPATH:$NIXNOTE/lib/commons-compress-1.2.jar
NN_CLASSPATH=$NN_CLASSPATH:$NIXNOTE/lib/commons-lang3-3.0.jar
NN_CLASSPATH=$NN_CLASSPATH:$NIXNOTE/lib/commons-logging-1.1.1.jar
NN_CLASSPATH=$NN_CLASSPATH:$NIXNOTE/lib/evernote-api-1.19.jar
NN_CLASSPATH=$NN_CLASSPATH:$NIXNOTE/lib/h2-1.3.158.jar
NN_CLASSPATH=$NN_CLASSPATH:$NIXNOTE/lib/httpclient-4.1.1.jar
NN_CLASSPATH=$NN_CLASSPATH:$NIXNOTE/lib/httpcore-4.1.jar
NN_CLASSPATH=$NN_CLASSPATH:$NIXNOTE/lib/httpmime-4.1.1.jar
NN_CLASSPATH=$NN_CLASSPATH:$NIXNOTE/lib/jaxen-1.1.3.jar
NN_CLASSPATH=$NN_CLASSPATH:$NIXNOTE/lib/jazzy.jar
NN_CLASSPATH=$NN_CLASSPATH:$NIXNOTE/lib/jtidy-r938.jar
NN_CLASSPATH=$NN_CLASSPATH:$NIXNOTE/lib/libthrift.jar
NN_CLASSPATH=$NN_CLASSPATH:$NIXNOTE/lib/log4j-1.2.14.jar
NN_CLASSPATH=$NN_CLASSPATH:$NIXNOTE/lib/pdfbox-app-1.6.0.jar
NN_CLASSPATH=$NN_CLASSPATH:$NIXNOTE/lib/poi-3.7-20101029.jar
NN_CLASSPATH=$NN_CLASSPATH:$NIXNOTE/lib/poi-ooxml-3.7.jar
NN_CLASSPATH=$NN_CLASSPATH:$NIXNOTE/lib/poi-ooxml-schemas-3.7-20101029.jar
NN_CLASSPATH=$NN_CLASSPATH:$NIXNOTE/lib/poi-scratchpad-3.7-20101029.jar
NN_CLASSPATH=$NN_CLASSPATH:$NIXNOTE/lib/tika.jar
NN_CLASSPATH=$NN_CLASSPATH:$NIXNOTE/lib/xmlbeans-2.3.0.jar
NN_CLASSPATH=$NN_CLASSPATH:$NIXNOTE/lib/xsdlib-20060615.jar

NN_CLASSPATH=$NN_CLASSPATH:$NIXNOTE/lib/qtjambi-linux32-4.5.2_01.jar
NN_CLASSPATH=$NN_CLASSPATH:$NIXNOTE/lib/qtjambi-linux32-gcc-4.5.2_01.jar
NN_CLASSPATH=$NN_CLASSPATH:$NIXNOTE/lib/qtjambi-linux64-4.5.2_01.jar
NN_CLASSPATH=$NN_CLASSPATH:$NIXNOTE/lib/qtjambi-linux64-gcc-4.5.2_01.jar

###################
# Run the program #
###################
cd $NIXNOTE
java -Xmx$NN_XMX -Xms$NN_XMS -XX:NewRatio=$NN_NEW_RATIO $NN_GC_OPT $NN_DEBUG -classpath $NN_CLASSPATH cx.fbn.nevernote.NeverNote --name=$NN_NAME
cd -
